parquet-column/src/main/java/org/apache/parquet/column/ParquetProperties.java

```diff
@@ -234,26 +280,139 @@ public ByteBufferAllocator getAllocator() {
+    private int maxRowCountForPageSizeCheck = DEFAULT_MAXIMUM_RECORD_COUNT_FOR_CHECK;
+    private boolean estimateNextSizeCheck = DEFAULT_ESTIMATE_ROW_COUNT_FOR_PAGE_SIZE_CHECK;
+    private ByteBufferAllocator allocator = new HeapByteBufferAllocator();
+
+    private Builder() {
+    }
+
+    private Builder(ParquetProperties toCopy) {
+      this.enableDict = toCopy.enableDictionary;
+      this.dictPageSize = toCopy.dictionaryPageSizeThreshold;
+      this.writerVersion = toCopy.writerVersion;
+      this.minRowCountForPageSizeCheck = toCopy.minRowCountForPageSizeCheck;
+      this.maxRowCountForPageSizeCheck = toCopy.maxRowCountForPageSizeCheck;
+      this.estimateNextSizeCheck = toCopy.estimateNextSizeCheck;
+      this.allocator = toCopy.allocator;
+    }
+
+    /**
+     * Set the Parquet format page size.
+     *
+     * @param pageSize an integer size in bytes
+     * @return this builder for method chaining.
+     */
+    public Builder withPageSize(int pageSize) {
+      Preconditions.checkArgument(pageSize > 0,
+          "Invalid page size (negative): %s", pageSize);
+      this.pageSize = pageSize;
+      return this;
+    }
+
+    /**
+     * Enable or disable dictionary encoding.
+     *
+     * @param enableDictionary whether dictionary encoding should be enabled
+     * @return this builder for method chaining.
+     */
+    public Builder withDictionaryEncoding(boolean enableDictionary) {
+      this.enableDict = enableDictionary;
+      return this;
+    }
+
+    /**
+     * Set the Parquet format dictionary page size.
+     *
+     * @param dictionaryPageSize an integer size in bytes
+     * @return this builder for method chaining.
+     */
+    public Builder withDictionaryPageSize(int dictionaryPageSize) {
+      Preconditions.checkArgument(dictionaryPageSize > 0,
+          "Invalid dictionary page size (negative): %s", dictionaryPageSize);
+      this.dictPageSize = dictionaryPageSize;
+      return this;
+    }
+
+    /**
+     * Set the {@link WriterVersion format version}.
+     *
+     * @param version a {@code WriterVersion}
+     * @return this builder for method chaining.
+     */
+    public Builder withWriterVersion(WriterVersion version) {
+      this.writerVersion = version;
+      return this;
+    }
+
+    public Builder withMinRowCountForPageSizeCheck(int min) {
+      Preconditions.checkArgument(min > 0,
+          "Invalid row count for page size check (negative): %s", min);
+      this.minRowCountForPageSizeCheck = min;
+      return this;
+    }
+
+    public Builder withMaxRowCountForPageSizeCheck(int max) {
+      Preconditions.checkArgument(max > 0,
+          "Invalid row count for page size check (negative): %s", max);
+      this.maxRowCountForPageSizeCheck = max;
+      return this;
+    }
+
+    // Do not attempt to predict next size check.  Prevents issues with rows that vary significantly in size.
+    public Builder estimateRowCountForPageSizeCheck(boolean estimateNextSizeCheck) {
```

Comment:
`withEstimate` implies that you're setting an estimate because it makes estimate a noun. Using estimate as the verb makes the boolean parameter make sense: `estimateRowCountForPageSizeCheck(false)`